File: Java/src/main/java/com/gildedrose/CheeseItem.java
package com.gildedrose;

/**
 *
 * @author markd
 */
public class CheeseItem extends GenericItem{
    
    public CheeseItem(String name, int sellIn, int quality) {
        super(name, sellIn, quality);
        QUALITY_CHANGE_BEFORE_SELLBY = 1;
        QUALITY_CHANGE_AFTER_SELLBY = 2;
    }
}

File: cpp-catch2/src/CheeseItem.cpp
//
// Created by markd on 3/9/2021.
//

#include "CheeseItem.h"

CheeseItem::CheeseItem(string name, int sellIn, int quality) : GenericItem(name, sellIn, quality) {
    QUALITY_CHANGE_BEFORE_SELLBY = 1;
    QUALITY_CHANGE_AFTER_SELLBY = 2;
}
